### PR TITLE
feat(brakes): better brakes simulation

### PR DESCRIPTION
**Pull request by Gurgel100**
Fixes #[issue_no]

## Summary of Changes

Move the simulation of the brake temperature into Rust. Introduces a more realistic simulation using physical principles. 

**A32NX**:
For the A32NX the brake temperature now is reported by the BSCU via ARINC429. Additionally the brake fans are now connected to the correct LGCIU signal and are consuming power from the correct electrical buses.

**A380X**:
Adds brake temperature simulation to the A380X but it's not used in the SD page yet.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. Check if the temperatures after normal landing make sense/are in the same range as in the current version.
1. Check if the temperatures after aborted takeoff are in the same range as in the current version.
1. Check that the brake fan doesn't work when the left main gear is in the air or LGCIU 2 is failed.
1. Check that the brake fan doesn't work when either DC 2 or AC 2 are not powered.
1. Check that the brake temperatures are not reported on the WHEEL page when DC 1 is not powered (amber X).

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs [hunk 1]
```diff
@@ -3882,6 +3976,22 @@ impl SimulationElement for A320HydraulicBrakeSteerComputerUnit {
...
             Ratio::new::<ratio>(reader.read(&self.autopilot_nosewheel_demand_id));
     }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        for (id, temp) in self
+            .brake_temperature_ids
+            .iter()
+            .zip(self.brake_temperatures)
+        {
+            // TODO: send status when not powered
```
Comment (tracernz) [inline fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs:3986]: Should send FW with FIXME to change to "no transmission" when we get around to implementing an A429 bus simulation. The values should also be quantised (steps of 5) and limited to the appropriate range (0 to 995), as the A429 encoding doesn't currently do that for us.

Comment (Gurgel100) [reply fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs:3986]: Since currently it's not simulated that the bscu is powered I added a comment.

**Review by 2hwk — APPROVED** [review]

**Comment by 2hwk** [comment]
Needs testing instructions

**Comment by 2hwk** [comment]
Quality Assurance Report

Discord Username  : 2cas
Object of testing : #9062
Aircraft          : A32NX
Tier of Testing   : 1
Date              : 19/10/2024

Testing Process: 

 - [x] Check if the temperatures after normal landing make sense/are in the same range as in the current version.
 - [x] Check if the temperatures after aborted takeoff are in the same range as in the current version.
 - [ ] Check that the brake fan doesn't work when the left main gear is in the air or LGCIU 2 is failed.
 - [ ] Check that the brake fan doesn't work when either DC 2 or AC 2 are not powered.
 - [x] Check that the brake temperatures are not reported on the WHEEL page when DC 1 is not powered (amber X).


Negatives:
 - Brake fan doesn't appear to work at all
 
Testing Results: 
Not Passed

**Comment by Gurgel100** [comment]
@2hwk thank you for your tests. I fixed the issue with the brake fan. Also I made some adjustment to the cooling of the brakes and the behavior of the temperature probes.

**Comment by tracernz** [comment]
Needs 0.13 changelog.

**Comment by 2hwk** [comment]
Quality Assurance Report

Discord Username  : 2cas
Object of testing : #8793
Aircraft          : A32NX
Tier of Testing   : 1
Date              : 29/10/2024

Testing Process: 

 - [ ] Check if the temperatures after normal landing make sense/are in the same range as in the current version.
 - [ ] Check if the temperatures after aborted takeoff are in the same range as in the current version.
 - [ ] Check that the brake fan doesn't work when the left main gear is in the air or LGCIU 2 is failed.
 - [ ] Check that the brake fan doesn't work when either DC 2 or AC 2 are not powered.
 - [x] Check that the brake temperatures are not reported on the WHEEL page when DC 1 is not powered (amber X).

image


Negatives:
 - Wheel temperatures continue increasing even after braking has stopped and plane is stationary. Unable to test brake fans as a result.
 
Testing Results: 
Not Passed

**Comment by Gurgel100** [comment]
How did you manage that? Interesting.

**Comment by beheh** [comment]
@Gurgel100 just to check, what's the status of this? Are you happy to proceed as-is, or is there something to be fixed after 2cas recent QA test?

**Comment by beheh** [comment]
If you need some data to tune this, there's a good document over at https://www.code7700.com/pdfs/carbon_brakes_airbus.pdf

**Comment by frenchairbuspilot** [comment]
QA Trainee Pilot Report

Discord          : jp320
Object of testing: #8793 
Tier of Testing  : 1
Date             : 11/11/2024

Testing Process: 
- Check if the temperatures after normal landing make sense/are in the same range as in the current version.
Failed, took off with brakes at 10°C, when landing gear out on appr brakes were at 125°, ldg 52T idle rev autobrake low, after vacating temp rapidly rose to 995 within 30sec.

- Check if the temperatures after aborted takeoff are in the same range as in the current version.
Failed, spawning at 10°C , messing around with the test below with elec, after 5 minutes brakes were showing 125°C ( i didn't move) after reject at 130kts, brakes went to 995 degrees in less than 20 sec.

- Check that the brake fan doesn't work when the left main gear is in the air or LGCIU 2 is failed.
I didn't check

- Check that the brake fan doesn't work when either DC 2 or AC 2 are not powered.
Failed, brakes fans still heard with DC 2 OFF, AC 2 OFF, and DC 2 + AC 2 OFF

- Check that the brake temperatures are not reported on the WHEEL page when DC 1 is not powered (amber X).
Working as expected.


Testing Results: 
Failed

Media:
Capture d’écran 2024-11-11 111604
Capture d’écran 2024-11-11 111818

**Comment by 2hwk** [comment]
Quality Assurance Report

Discord Username  : 2cas
Object of testing : #8793
Aircraft          : A32NX
Tier of Testing   : 1
Date              : 16/11/2024

Testing Process: 

 - [x] Check if the temperatures after normal landing make sense/are in the same range as in the current version.
 - [x] Check if the temperatures after aborted takeoff are in the same range as in the current version.
 - [x] Checked brake fan operation
    - [x] Sensor reported temperature decreases when brake fan is active
    - [x] Toggled brake fan off, reported temperature rose but did not reach previous peak of 330
    - [x] Toggled brake fan back on, reported temperature dropped down to 150
    - [x] Toggled brake fan off. Temperature rose to 260 and recorded as dropping on its own without the brake fan (after a few minutes)
 - [x] Check that the brake fan doesn't work when the left main gear is in the air or LGCIU 2 is failed.
 - [x] Check that the brake fan doesn't work when either DC 2 or AC 2 are not powered.
 - [x] Check that the brake temperatures are not reported on the WHEEL page when DC 1 is not powered (amber X).


Negatives:
 - BRK FAN memo still shows when BRK FAN is not operating because of LGCIU 2/DC 2/AC 2 outage, is this correct?
 - Brakes fan sound can still be heard in exterior when LGCIU 2 Power, DC 2 and/or AC 2 are off
 - Sensor temperatures however are now correct.

image


Testing Results: 
QA Passed

Conclusions:
Some minor issues left but this PR has been long in the teeth, patch PR is a good idea.

**Pull request merged**